Improve #12008 UX, clear scrollback by default on `clear`

T3sT3ro opened the pull request:
<!--
if this PR closes one or more issues, you can automatically link the PR with
them by using one of the [*linking keywords*](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword), e.g.
- this PR should close #xxxx
- fixes #xxxx

you can also mention related issues, PRs or discussions!
-->

# Description
<!--
Thank you for improving Nushell. Please, check our [contributing guide](../CONTRIBUTING.md) and talk to the core team before making major changes.

Description of your pull request goes here. **Provide examples and/or screenshots** if your changes affect the user experience.
-->

Related to #11693. It looks like there is no reason for Nu shell's `clear` to behave differently than other shells' `clear`. To improve the UX and fulfill the user expectations, the default has been adjusted to work the same as in other shells by clearing the scrollback buffer. For edge cases where someone depends on the current behavior of keeping the scrollback, a `-k` option has been added.

# User-Facing Changes
<!-- List of all changes that impact the user experience here. This helps us keep track of breaking changes. -->

- Improve the UX of `clear` by changing the default behavior to the same as other popular shells, i.e clear scrollback by default.
- Remove `-a --all` flag, make it the default behavior to clear the scrollback
- Add `-k --keep-scrollback` flag for backward compat to keep the scrollback buffer

# Tests + Formatting
<!--
Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass (on Windows make sure to [enable developer mode](https://learn.microsoft.com/en-us/windows/apps/get-started/developer-mode-features-and-debugging))
- `cargo run -- -c "use toolkit.nu; toolkit test stdlib"` to run the tests for the standard library

> **Note**
> from `nushell` you can also use the `toolkit` as follows
> ```bash
> use toolkit.nu  # or use an `env_change` hook to activate it automatically
> toolkit check pr
> ```
-->

This is a simple change flipping the flag and default behavior, no tests should be needed.

# After Submitting
<!-- If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date. -->

- [ ] update the `clear` command docs

NotTheDr01ds commented:
Can't comment on unedited code via GitHub code-review, but:

* Recommend adding an `extra_description` that reads:

  > By default clears the current screen and the off-screen scroll-back buffer

NotTheDr01ds commented on crates/nu-command/src/platform/clear.rs:
```diff
@@ -57,8 +57,8 @@ impl Command for Clear {
             Example {
-                description: "Clear the terminal and its scroll-back history",
-                example: "clear --all",
+                description: "Clear the terminal without its scroll-back history",
```
```suggestion
                description: "Clear the terminal but not its scroll-back history",
```

NotTheDr01ds commented:
> update the clear command docs

Funny - I said the same thing on my first PR ;-)

But the docs are auto-generated from the help that we're modifying here, so it gets taken care of automatically.

T3sT3ro commented:
I added the `extra_description` just like you suggested. I also changed "scroll-back" to "scrollback" after checking for usages of both, as the second one is more common wherever I searched.

> But the docs are auto-generated from the help that we're modifying here, so it gets taken care of automatically.

I suspected this might be the case but didn't have the time to verify if that's true. I planned to work on that next if it wasn't :P

fdncred commented on crates/nu-command/src/platform/clear.rs:
```diff
@@ -57,8 +61,8 @@ impl Command for Clear {
-                description: "Clear the terminal and its scroll-back history",
-                example: "clear --all",
+                description: "Clear the terminal but not its scrollback history",
+                example: "clear -k",
```
We generally like to stay verbose in examples.
```suggestion
                example: "clear --keep-scrollback",
```

fdncred commented:
Thanks. Let's give 'er a try.
